src/engine/renderer/tr_shade.cpp

```diff
@@ -50,15 +50,20 @@ static void GLSL_InitGPUShadersOrError()
 	// standard light mapping
 	gl_shaderManager.load( gl_lightMappingShader );
 
+	// Deprecated forward renderer uses r_dynamicLight -1
+	// Dynamic shadowing code also needs this shader.
```

Comment:
To me it seems likely that it's a mistake that the code using the shader even runs (like there should have been an if (r_dynmaicLight == -1 around it). The comment could say something like "The shadowing code also accesses this shader. Possibly, that's a bug and it should only be used when the forward renderer is enabled."

Reply:
That was my thought and I was gonna add another `r_dynamicLight < 0` test until I see @gimhael comment saying in https://github.com/DaemonEngine/Daemon/commit/a09f03bc8e775d83ac5e057593eff4e88cdea7eb

> This re-enables shadows for players in the tiled renderer

So it looks like this was intentional.

My priority is to fix the crash, I haven't seen shadows working in years anyway but we should not crash. I'll add a comment.

Reply:
Maybe the change is intended to make cg_shadows 1 work in the tiled renderer but not higher values.

Reply:
That's a good point!

On the other hand a09f03bc8e775d83ac5e057593eff4e88cdea7eb also modify light stuff, and if I'm right `cg_shadows 1` (blob shadows) are unrelated to lights, they're just a darker blob painted at player's feet…

Reply:
OK, LGTM

Reply:
I added a comment anyway.